biomage/experiment/pull.py

```diff
@@ -52,16 +73,16 @@ def update_config_if_needed(file, table_name, experiment_id):
         remote_cfg = remove_key(remote_cfg, "pipeline")
 
     # if the local config was not found or it's different from the remote => update
-    if not found or DeepDiff(local_cfg, remote_cfg):
-        save_cfg_file(remote_cfg, file)
-        Summary.add_changed_file(file)
+    if not found or local_cfg != remote_cfg:
```

Comment:
why is there no need for this DeepDiff anymore?

Reply:
DeepDiff was not considering equal fields with same value but different type (educated guess) considering different `Decimal(10)` and just `10` and thus failing the comparison. However, just using equal works fine (the structures are saved / loaded the same way so it seems they can indeed be compared like that).

Reply:
you can also remove it from requirements.txt in this case too: https://github.com/biomage-ltd/biomage-utils/blob/master/requirements.txt#L16